src/main/java/com/google/devtools/build/lib/remote/http/HttpCacheClient.java

```diff
@@ -774,6 +808,19 @@ public void close() {
     return null;
   }
 
+  private File createCompressedUpload(String name, InputStream input) throws IOException {
+    File compressedOutput = File.createTempFile(name, ".gz");
+    compressedOutput.deleteOnExit();
```

Comment:
Nit: this might not have the desired effect as bazel runs with a persistent server

Reply:
Agreed, this is just a cautious fallback in case of JVM shutdown before the request completes.